fenix/app/src/main/java/org/mozilla/fenix/translations/TranslationsDialogFragment.kt

```diff
@@ -126,14 +164,24 @@ class TranslationsDialogFragment : BottomSheetDialogFragment() {
                                         )
                                     },
                                     onTranslateButtonClick = {
-                                        interactor.onTranslate(
-                                            tabId = args.sessionId,
-                                            fromLanguage = null,
-                                            toLanguage = null,
-                                            null,
-                                        )
+                                        fromSelected?.code?.let { fromLanguage ->
+                                            toSelected?.code?.let { toLanguage ->
+                                                TranslationsAction.TranslateAction(
```

Comment:
Will follow-up on null case in [bug 1875534](https://bugzilla.mozilla.org/show_bug.cgi?id=1875534)

Reply:
Optionally, this is so deeply nested that I would consider defining a lambda at (or before, even) the `setContent` level. This could also wait until the null case bug though

Reply:
Thanks! Added as a comment to make this refactor in the null case [bug 1875534](https://bugzilla.mozilla.org/show_bug.cgi?id=1875534).

Reply:
I think the nullability can be handled y the button state. The translate button should not be enabled when one or both languages have no selection so it should not even end up being handled here (`onTranslateButtonClick`). That would allow removing nesting and having this whole block as:

```
                                    onTranslateButtonClick = {
                                        browserStore.dispatch(
                                            TranslationsAction.TranslateAction(
                                                tabId = args.sessionId,
                                                fromLanguage = fromSelected?.code!!,
                                                toLanguage = toSelected?.code!!,
                                                options = null,
                                            ),
                                        )
                                    },
```
Any thoughts against this? This would enforce the assumption no null values would end up being used in this lambda, and we do want to catch cases where this "contract" is broken.